src/alerts.js

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* eslint-disable sorting/sort-object-props */
+export const ALERT_LEVEL = {
+  WARNING: 1,
+  FATAL: 2
+};
+
+export const ALERT_DESCRIPTION = {
+  CLOSE_NOTIFY: 0,
+  UNEXPECTED_MESSAGE: 10,
+  BAD_RECORD_MAC: 20,
+  RECORD_OVERFLOW: 22,
+  HANDSHAKE_FAILURE: 40,
+  ILLEGAL_PARAMETER: 47,
+  DECODE_ERROR: 50,
+  DECRYPT_ERROR: 51,
+  PROTOCOL_VERSION: 70,
+  INTERNAL_ERROR: 80,
+  USER_CANCELLED: 90,
+  MISSING_EXTENSION: 109,
+  UNSUPPORTED_EXTENSION: 110,
+  UNKNOWN_PSK_IDENTITY: 115,
+  NO_APPLICATION_PROTOCOL: 120,
+};
+/* eslint-enable sorting/sort-object-props */
+
+function alertTypeToName(type) {
+  for (const name in ALERT_DESCRIPTION) {
+    if (ALERT_DESCRIPTION[name] === type) {
+      return name;
+    }
+  }
+  return 'UNKNOWN';
+}
+
+
+export class TLSError extends Error {
+  constructor(description = ALERT_DESCRIPTION.INTERNAL_ERROR) {
+    super(`TLS Error: ${alertTypeToName(description)}`);
+    this.description = description;
+    switch (description) {
+      case ALERT_DESCRIPTION.CLOSE_NOTIFY:
+      case ALERT_DESCRIPTION.USER_CANCELLED:
+        this.level = ALERT_LEVEL.WARNING;
```

Comment:
I'm not using it, and only included it because it was mentioned alongside `close_notify` in [1].  I'll be very happy to remove it.

[1] https://tools.ietf.org/html/rfc8446#section-6.1